index.js

```diff
@@ -24,14 +25,26 @@ module.exports = function frameguard (options) {
   }
 
   if (directive === 'ALLOW-FROM') {
-    if (!isString(domain)) {
-      throw new Error('X-Frame: ALLOW-FROM requires a second parameter')
+    if (!isString(domain) && !isArray(domain)) {
+      throw new Error('X-Frame: ALLOW-FROM requires a domain parameter')
     }
-    directive = 'ALLOW-FROM ' + domain
+  }
+
+  function header (req) {
+    var headerValue = directive
+    if (directive === 'ALLOW-FROM' && isString(domain)) {
+      headerValue = 'ALLOW-FROM ' + domain
+    }
+
+    if (directive === 'ALLOW-FROM' && isArray(domain)) {
+      headerValue = 'ALLOW-FROM ' + (domain.indexOf(req.hostname) >= 0 ? req.hostname : domain[0])
+    }
+
+    return headerValue
   }
 
   return function frameguard (req, res, next) {
-    res.setHeader('X-Frame-Options', directive)
+    res.setHeader('X-Frame-Options', header(req))
```

Comment:
Could you move the contents of `header` into this middleware? No need to have an extra function.